lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -65,23 +107,51 @@ pub mod aggregator {
     }
 
     #[pyfunction]
-    #[pyo3(signature = (name, status, tags, hostname, message=None))]
-    fn submit_service_check(name: String, status: i32, tags: Vec<String>, hostname: String, message: Option<String>) {
-        println!(
+    fn submit_service_check(
+        _class: PyObject, _check_id: String, name: String, status: i32, tags: Vec<String>, hostname: String,
+        message: Option<String>,
+    ) {
+        trace!(
             "submit_service_check called with name: {}, status: {}, tags: {:?}, hostname: {}, message: {:?}",
-            name, status, tags, hostname, message
+            name,
+            status,
+            tags,
+            hostname,
+            message
         );
-    }
 
-    #[pyfunction]
-    fn metrics(name: String) -> Vec<String> {
-        println!("metrics called for: {}", name);
-        vec![] // Dummy return
+        if let Ok(service_check_status) = status.try_into() {
+            let mut tag_set = TagSet::default();
+            for tag in tags {
+                tag_set.insert_tag(tag);
+            }
+
+            let service_check = ServiceCheck::new(name.clone(), service_check_status)
+                .with_tags(tag_set.into_shared())
+                .with_hostname(if hostname.is_empty() {
+                    None
+                } else {
+                    Some(hostname.into())
+                })
+                .with_message(Some(message.unwrap_or_default().into()));
```

Comment:
This should probably be like... `message.map(Into::into)` to do the conversion from `Option<String>` to `Option<MetaString>`.